### PR TITLE
Add contribution guidelines under .github

### PR DESCRIPTION
**Pull request by niboshi**
This PR is for #2750. (And repost of PR #2763)

* [CONTRIBUTING.md](https://github.com/pfnet/chainer/blob/_v2-dot-github/.github/CONTRIBUTING.md)
* [ISSUE_TEMPLATE.md](https://github.com/pfnet/chainer/blob/_v2-dot-github/.github/ISSUE_TEMPLATE.md)
* [PULL_REQUEST_TEMPLATE.md](https://github.com/pfnet/chainer/blob/_v2-dot-github/.github/PULL_REQUEST_TEMPLATE.md)

See also: https://github.com/blog/1184-contributing-guidelines

Also added `ISSUE_TEMPLATE.md` and `PULL_REQUEST_TEMPLATE.md`.

### CODE REVIEW

**Comment by delta2323** [comment]
I am thinking of merging this PR after updating the contribution guide for v2 (working as #2773). Could you update `CONTRIBUTING.md` accordingly after the PR is merged.

**Comment by niboshi** [comment]
OK!

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+Some GPU tests involve multiple GPUs.
+If you want to run GPU tests with insufficient number of GPUs, specify the number of available GPUs by ``--eval-attr='gpu<N'`` where ``N`` is a concrete integer.
+For example, if you have only one GPU, launch ``nosetests`` by the following command to skip multi-GPU tests:
+
+```shell
+$ nosetests path/to/gpu/test.py --eval-attr='gpu<2'
+```
+
+Some tests spend too much time.
```
Comment (bkvogel) [inline .github/CONTRIBUTING.md]: There are some tests that take a long time to complete.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,296 @@
+Chainer Contribution Guide
+==========================
+
+This is a guide for all contributions to Chainer.
+Anyone that wants to register an issue or to send a pull request should read through this document.
```
Comment (bkvogel) [inline .github/CONTRIBUTING.md]: Consider rewording:
The core developers welcome and appreciate any contributions to Chainer. This document describes the various types of contributions and the procedures that we suggest to follow. Anyone that would like to register an issue or to send a pull request should first read through this document.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+They are symbols imported by packages and sub-packages of ``chainer``.
+For example, ``chainer.Variable`` is a shortcut of ``chainer.variable.Variable``.
+**It is not allowed to use such shortcuts in the ``chainer`` library implementation**.
+Note that you can still use them in ``tests`` and ``examples`` directories.
+Also note that you should use shortcut names of CuPy APIs in Chainer implementation.
+
+Once you send a pull request, your coding style is automatically checked by [Travis-CI](https://travis-ci.org/pfnet/chainer/).
+The reviewing process starts after the check passes.
+
+The CuPy is designed based on NumPy's API design. CuPy's source code and documents contain the original NumPy ones.
```
Comment (bkvogel) [inline .github/CONTRIBUTING.md]: The CuPy API is designed

**Comment by bkvogel** [comment]
I added a few optional wording change suggestions.

**Comment by niboshi** [comment]
Thank you for review!
As we discussed internally, the content of CONTRIBUTING.md will be a link to the official contribution guide (or vise versa).
The official guide is being composed in #2773.

**Comment by delta2323** [comment]
LGTM

**Pull request merged**